openedx/adg/lms/applications/api/serializers.py

```diff
@@ -103,3 +104,25 @@ class Meta:
         read_only_fields = (
             'id', 'created', 'modified',
         )
+
+
+class ReferenceSerializer(serializers.ModelSerializer):
+    """
+    Serializer for the `Reference` model
+    """
+
+    def validate_user_application(self, value):
+        """
+        Validate that max reference limit against a user application is not exceeded
+        """
+        if self.instance:
+            return value
```

Comment:
Let's please describe as add a comment here, or in the doc string that this is to identify if the call is for creation or any other operations

Reply:
I thought about adding a comment here, but don't you think that it is quite obvious by looking at the code that if an instance is provided to the serializer, it means that it is an update or delete operation, and if an instance is not provided, it is of course a new creation (POST) operation. We also have adding an explanatory docstring, so I feel an extra comment is not needed. As per clean code guidelines, we should avoid code comments as much as possible and strive towards making our code self-explanatory. I feel this is explanatory enough, as we use this instance approach in many other places as well, like forms, etc.

But of course, I'm no the right person to judge this, since I've written this code. So if you still feel that we require a comment, I'll be more than happy to add one.

Reply:
@MunamMubashir-arbisoft as a reviewer, would you like to share your thoughts on this?

Reply:
I think we should add a line in the docstring to better communicate the method's function (no pun intended). A single line to explain it similar to how you explained it Ahmed bhai.

Reply:
@harris-irfan actually it is not that self explanatory at least readers would have to think about it for some time before they get it, if we really want to make it self explanatory without adding a comment, then we can write a private method here, `def is_create_request` and check is instance available or not in that method and here we will simply add a check

```
if is_create_request() and Reference.objects.filter(user_application=value).count() == MAX_NUMBER_OF_REFERENCES:         
           raise serializers.ValidationError(MAX_REFERENCE_ERROR_MSG)
```
This long line could also be improved by defining a variable above.

Reply:
@ahmed-faraz46 I've updated the docstring. If it suffices, good, otherwise I'll do as you mentioned.